### ISSUE TITLE
[BUG] Predict N-dimensional data with MLmodel

### ISSUE THREAD

**Issue by aseltmann** [mlflow/mlflow#2830]
### Willingness to contribute
The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the MLflow code base?

- [ ] Yes. I can contribute a fix for this bug independently.
- [ ] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
- [ x] No. I cannot contribute a bug fix at this time.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Manjaro Kernel: 5.5.19-1-MANJARO
- **MLflow installed from (source or binary)**: binary
- **MLflow version (run ``mlflow --version``)**: 1.8.0
- **Python version**: 3.8.2
- **npm version, if running the dev UI**:
- **Exact command to reproduce**: 

### Describe the problem
I have trained a model using `tensorflow.keras` and logged the training using `mlflow.tensorflow.autolog(every_n_iter=1)`. My training data and test data comes as tf.Datasets. I construct the model using the TensorFlow Functional API. The model is serialisable and I am saving it using `model.save('data/exp-devtest/unet.tf', save_format='tf')`. Mlflow successfully logs the model (see other info / logs). I am having troubles to somehow load this MLmodel and get predictions from my data with it.

### Code to reproduce issue
Train a model
import sys
import mlflow
import mlflow.tensorflow
import numpy as np
import tensorflow as tf

mlflow.tensorflow.autolog(every_n_iter=1)

data = np.array([[[5.1], [5.9], [6.9]], 
                 [[3.3], [3.0], [3.1]],
                 [[1.7], [4.2], [5.4]],
                 [[0.5], [1.5], [2.1]]])
labels = np.array([[[0], [1], [1]],
                   [[1], [1], [0]],
                   [[0], [1], [0]],
                   [[0], [0], [1]]])
print(data.shape, labels.shape) # (4, 3, 1) (4, 3, 1)

data_tensor = tf.convert_to_tensor(value=data)
labels_tensor = tf.convert_to_tensor(value=labels)
labels_tensor = tf.cast(labels_tensor, tf.float32)

dataset = tf.data.Dataset.from_tensor_slices((data_tensor, labels_tensor))
dataset = dataset.shuffle(buffer_size=1000).repeat().batch(batch_size=2)

### Other info / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.
```yaml
>>> ls $model_path
    conda.yaml  data/  MLmodel

>>> ls $model_path/data
    keras_module.txt  model.h5

>>> cat $model_path/MLmodel
    artifact_path: model
    flavors:
      keras:
        data: data
        keras_module: tensorflow.keras
        keras_version: 2.2.4-tf
      python_function:
        data: data
        env: conda.yaml
        loader_module: mlflow.keras
        python_version: 3.8.2
    run_id: 52764cca60c64d9bbdbf89293ef01f48
    utc_time_created: '2020-05-14 14:36:15.881485'

>>> cat $model_path/conda.yaml
    channels:
    - defaults
    dependencies:
    - python=3.8.2
    - pip
    - pip:
      - mlflow
      - tensorflow==2.2.0-dev20200506
    name: mlflow-env

>>> cat $model_path/data/keras_module.txt
    tensorflow.keras
```

### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

**Comment by MainRo** [comment]
I have the same issue. The conversion of the result from numpy to a pandas dataframe prevents using results in more than 2d. Meanwhile the predict method accepts input data in more than 2d.
Can we just remove the pd.DataFrame conversion ? 
It seems that the serve functions accept numpy data: 
https://github.com/mlflow/mlflow/blob/74be64c8c469aa6cce5cebb92fcafea4bd5ad3a9/mlflow/pyfunc/scoring_server/__init__.py#L245

I suppose this change would allows mlflow serve to still work while allow using > 2d results in custom serving tools.

**Comment by mohammedi-haroune** [comment]
In fact, many issues were open in the mlflow github repo to allow predict to accept multidimensional arrays:

Open issues: 
* [[BUG] ONNX predict fails for high-dimensional models #3229](https://github.com/mlflow/mlflow/issues/3229)
* [[BUG] Predict N-dimensional data with pyfunc from MLmodel #2830](https://github.com/mlflow/mlflow/issues/2830)
* [[FR] Support tensor inputs to MLflow inference APIs #3159](https://github.com/mlflow/mlflow/issues/3159)

Pull requests:
* [Added support for high dimensional onnx models #3307](https://github.com/mlflow/mlflow/pull/3307/)

Closed issues:
* [How to pass 2D array as an input to the MLFlow Model #1480](https://github.com/mlflow/mlflow/issues/1480)
* [[FR]How to transfer the multi-dimensional pixel information of the image to the model after the model is deployed as docker？ #1668](https://github.com/mlflow/mlflow/issues/1668)
* [[BUG] Model predict with json input fails if the dataframe has multi-index #2412](https://github.com/mlflow/mlflow/issues/2412)
* [[FR] Mlflow serve - sending an image #2452](https://github.com/mlflow/mlflow/issues/2452)
* [Recommended handling of I/O for pyfunc? #702](https://github.com/mlflow/mlflow/issues/702)
* [TensorFlow wrapper ravels outputs on predict #1066](https://github.com/mlflow/mlflow/issues/1066)

This is clearly something that blocks so many users from properly use their models logged by MLFlow, some work was done to allow onnx to accept multidimensional data in the pull request linked above trying to understand if the client wants to shape their DataFrame as a multi-dimensional array. I still don't understand why MLFlow predict method is tightly coupled to DataFrames which clearly can't represent a wide range of inputs, why don't we just allow passing numpy arrays ?
